4-1.cpp

```diff
@@ -0,0 +1,151 @@
+#include <iostream>
+#include <cstdlib>
+
+using namespace std;
+
+// Вывод массива
+// iSize - размер массива
+// piArray - указатель на массив
+void vGetArray(int iSize, int* piArray) {
```

Comment:
1. Это НЕ документация. Это хорошие комментарии.
Для документрирования кода необходимо использовать [Doxygen](https://www.doxygen.nl/manual/docblocks.html). Документация фунций должна располагаться ДО, НО НИКАК НЕ ПОСЛЕ (И НЕ ВНУТРИ) [подробнее см. пример с enum EnumType](https://www.doxygen.nl/manual/docblocks.html)
2. На лекции 10.12. 2021 было прочитано, какой тип данных использовать для размера массива и почему! 
Для индексации массивов и работы с их размерностью лучше использовать тип [`size_t`](https://en.cppreference.com/w/cpp/types/size_t). Для проверки корректности значения, которое будет использовано в качестве размера массива, нужно использовать тип [`int`](https://en.cppreference.com/w/cpp/language/types#Signed_and_unsigned_integer_types), например, следующим образом:

```cpp
size_t get_size(const std::string& message)
{
    int size = -1;
    std::cout << message;
    std::cin >> size;

    if (size < 0)
    {
        throw std::out_of_range("Incorrect size. Value has to be greater or equal zero.");
    }

    return size;
}
```
4.
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.